flink-filesystems/flink-s3-fs-presto/src/test/java/org/apache/flink/fs/s3presto/PrestoS3FileSystemITCase.java

```diff
@@ -143,7 +157,20 @@ public void testDirectoryListing() throws Exception {
 			fs.delete(directory, true);
 		}
 
-		// now directory must be gone
-		assertFalse(fs.exists(directory));
+		// now directory must be gone (this is eventually-consistent, though!)
+		checkPathExists(fs, directory, false, deadline);
+	}
+
+	private static void checkPathExists(
```

Comment:
Can we deduplicate this function?

Reply:
makes sense - I created `org.apache.flink.core.fs.FileSystemTestUtils` for this helper method